src/apprt/gtk-ng/class/search_bar.zig

```diff
@@ -0,0 +1,380 @@
+    }
+
+    // Called when user clicks the "previous result" button.
+    // This tells the connected terminal surface to move to the previous search match.
+    fn onPrevClicked(self: *Self, _: *gtk.Button) callconv(.C) void {
+        const priv = self.private();
+        if (priv.surface) |surface| {
+            // Send the search_previous action to the terminal.
+            // This will highlight the previous match and scroll to show it.
+            surface.performAction(.search_previous) catch |err| {
+                log.warn("failed to perform search_previous action: {}", .{err});
+            };
+        }
+    }
+
+    // Called when user clicks the "next result" button.
+    // This tells the connected terminal surface to move to the next search match.
+    fn onNextClicked(self: *Self, _: *gtk.Button) callconv(.C) void {
+        const priv = self.private();
+        if (priv.surface) |surface| {
+            // Send the search_next action to the terminal.
+            // This will highlight the next match and scroll to show it.
+            surface.performAction(.search_next) catch |err| {
+                log.warn("failed to perform search_next action: {}", .{err});
+            };
+        }
+    }
+
+    // Called when user clicks the close button or wants to exit search.
+    // This hides the search bar and returns the terminal to normal mode.
+    fn onCloseClicked(self: *Self, _: *gtk.Button) callconv(.C) void {
+        // Send a signal that the search bar should be closed.
+        // The parent window will listen for this and hide the search interface.
+        gobject.ext.impl_helpers.emitSignal(
+            self,
+            signals.search_closed.name,
+            .{},
+        );
+    }
+
+    // Called when user types text in the search input box.
+    // This triggers a new search with the updated text every time they type.
+    fn onSearchChanged(self: *Self, _: *gtk.Entry) callconv(.C) void {
+        const priv = self.private();
+        
+        // Get the current text from the search input box.
+        // This is what the user has typed so far.
+        const search_text = priv.search_entry.getText();
+        
+        // If we have a connected terminal surface, start searching.
+        // The surface will use its SearchManager to find matches in history.
+        if (priv.surface) |surface| {
+            surface.startSearch(search_text) catch |err| {
+                log.warn("failed to start search: {}", .{err});
+            };
+        }
+
+        // Send a signal with the new search text.
+        // Other parts of the app can listen for this and respond.
+        gobject.ext.impl_helpers.emitSignal(
+            self,
+            signals.search_text_changed.name,
+            .{search_text},
+        );
+    }
+
+    // Called when user presses keys while focused on the search input.
+    // This handles special keys like Enter (next result) and Escape (close search).
+    fn onKeyPressed(
```

Comment:
You can use a `gtk.SearchEntry` instead which has special signals specifically for this. (See command palette)